### DISCUSSION TITLE
How to automatically inject trace details to application logs written in Nodejs / Go

### DISCUSSION THREAD

**Discussion by jaisonsteephen** [open-telemetry/opentelemetry-js-api#67] [Q&A] [answered]
In python, there is an **option to inject trace details (traceid,spanid) automatically to application logs** using environment variable (export OTEL_PYTHON_LOG_CORRELATION=true). **Is there something similar in Nodejs** and Go. I couldn't find any autoinstrumentaion for tracing in Go (We can manually instrument using context, by creating span). So I don't expect answer from Go language, but probably, Node might have a solution

**Comment by SergeyKanzhelev** [comment]
Transferred to JS repository for @open-telemetry/javascript-approvers attention

**Comment by dyladan** [comment, marked as answer]
Depends what logging framework you're using. Some of them have autoinstrumentation packages that do exactly what you're saying.

- [bunyan](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-bunyan)
- [pino](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-pino)
- [winston](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-winston)

There is no such feature for direct console.log

**Reply by jaisonsteephen** [reply]
**Thank you very much**. It is working with bunyan.  Is it possible with **log4js as well ?**
Also interested to know, such provision in **Go**

**Reply by dyladan** [reply]
There is no log4js instrumentation i'm aware of, but if you would like to contribute one we would appreciate it.

You will have to ask in the go repo for that :)

**Reply by jaisonsteephen** [reply]
Thanks for the update. Will discuss your comments with my team, and will do the needful. thanks once again

**Reply by tigrannajaryan** [reply]
@jaisonsteephen if you would like to contribute a logging solution please read this spec https://github.com/open-telemetry/oteps/blob/main/text/logs/0150-logging-library-sdk.md
It explains how we would like the logging libraries to work in Otel. The currently existing implementation for bunyan is experimental and does not necessarily follow the spec (the implementation was done before the spec was written). We would like future contributions to follow the spec.

**Reply by jaisonsteephen** [reply]
Thanks for your replay. Sorry for the delay in response. I wish i should try this, but due to the busy project schedule &  being a fresher in Golang , currently I cannot due this. But thanks for this information. It may help us in future.
